### ISSUE TITLE
Wikimedia\\Rdbms\\DBQueryError: Error 1406: Data too long for column 'mod_comment' at row 1

### ISSUE THREAD

**Issue by nymphlode** [edwardspec/mediawiki-moderation#91]
I am a moderator for a wiki hosted by Miraheze. Recently, while trying to edit someone's entry before approving it, I ran into this error message:

- [99d5db9dc351cd0507f5811e] 2024-10-31 22:46:00: Fatal exception of type "Wikimedia\Rdbms\DBQueryError"

When I brought it up with Miraheze, they saw this error on the backend:

- [d7e6c66fa761d3ed53f6fd16] /wiki/Special:Moderation?modaction=editchangesubmit&modid=5194   Wikimedia\\Rdbms\\DBQueryError: Error 1406: Data too long for column 'mod_comment' at row 1

From this, I was able to find that there was too much text within the Summary box when I was trying to save the edit I made. It had to be shortened in order to go through.

I asked Miraheze if this issue has any chance of being resolved, and they told me to create a ticket here.

Let me know if I can add on any additional information to help 👍

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I have identified the issue and implemented a solution. The problem was that the 'mod_comment' column in the 'moderation' table had a limited size, causing the 'Data too long' error when trying to save longer edit summaries. To fix this, I've modified the SQL schema to increase the size of the 'mod_comment' column.

Here's what I've done:
1. Updated the SQL schema in the 'patch-moderation.sql' file to change the 'mod_comment' column type from VARCHAR(255) to TEXT.
2. Created a new SQL patch file to alter existing tables.
3. Updated the updater class to apply the new patch when upgrading the extension.

This change will allow for longer edit summaries without encountering the 'Data too long' error.

I also created pull request: https://github.com/edwardspec/mediawiki-moderation/pull/92


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
